homeassistant/components/heos/__init__.py

```diff
@@ -81,6 +81,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: HeosConfigEntry) -> bool
     if entry.unique_id is None:
         hass.config_entries.async_update_entry(entry, unique_id=DOMAIN)
 
+    # Migrate non-string device identifiers.
+    device_registry = dr.async_get(hass)
+    for device in device_registry.devices.get_devices_for_config_entry_id(
+        entry.entry_id
+    ):
+        identifier = next(iter(device.identifiers))
```

Comment:
I believe we want to double check that the identifier is indeed the one from HEOS

Reply:
Check added. I didn't realize `get_devices_for_config_entry_id` could return devices from other domains. 😬

Reply:
It doesn't, but the chance is there that devices are merged (with one from a network integration for example), but I don't think you have the mac address set up yet, but I rather be safe than sorry

Reply:
Oh that makes sense. Yeah, since there's no connections currently, I won't, but I agree, better to have it future-proof!